Automattic/wp-calypso
# Signup: Add Siteless support to Plans and Domains steps

#### Changes proposed in this Pull Request

* This is an updated version of #39704 , which was created a couple months ago by @mmtr 
* When the feature flag is turned on (it needs to be turned on manually right now), the site creation step will be skipped in the signup flow unless the users opt for a free plan.
* Instead of adding the flag checks directly in the fulfilled/apiRequest function for each step, two new steps were added called plans-siteless, domains-siteless, this is because we need to control in which flows we actually want to add siteless support without having a huge effect into the other steps for that flow.
* Part of #39424 and pbOQVh-cF-p2

#### Testing instructions

* Apply D43695-code and sandbox the API
* Go to `start/domains?flags=+signup/siteless-checkout/onboarding`
* Complete all the steps and select a paid plan.
* Make sure you're redirected to the checkout page and the site has not been created.
  * Please note that the checkout page won't work, this is because the needed changes are not covered in this PR and the backend changes are not ready yet.
* Make sure the signup flow remains unmodified (sites created before checkout) when turning off the feature flag by going to `/start/domains?flags=-signup/siteless-checkout/onboarding`

##### Here are the scenarios that I have validated for the **onboarding** flow.

**When signup/siteless-checkout/onboarding is on**
`/start/domains?flags=+signup/siteless-checkout/onboarding`
No Domain + Paid Plan = Go to siteless checkout
Paid Domain + Paid Plan = Go to siteless checkout
No Domain + Free Plan = Site Created + go to site home

**When signup/siteless-checkout/onboarding is off**
`/start/domains?flags=-signup/siteless-checkout/onboarding`
No Domain + Paid Plan = Site Created + Go to checkout
Paid Domain + Paid Plan = Site Created + Go to checkout
No Domain + Free Plan = Site Created + go to site home

#### Known Issues
* If you repeat the flow several times you will end up with multiple domains in the cart every time you land to the checkout page, I'll address that in a future PR.
